tests/components/colorSelector-test.js

```diff
@@ -0,0 +1,87 @@
+jest
+.dontMock('../../js/components/colorSelector')
+.dontMock('../../js/components/accessibleButton')
+.dontMock('../../js/components/higher-order/accessibleMenu')
+.dontMock('../../js/components/utils')
+.dontMock('../../js/constants/constants')
+.dontMock('classnames');
+
+var React = require('react');
+var ReactDOM = require('react-dom');
+var TestUtils = require('react-addons-test-utils');
+var ColorSelector = require('../../js/components/colorSelector');
+var AccessibleButton = require('../../js/components/accessibleButton');
+var CONSTANTS = require('../../js/constants/constants');
+
+describe('Tabs', function() {
+  var props, colors, node, tree, component, buttons;
+
+  // Using ReactDOM.render instead of test utils in order to allow re-render to simulate props update
+  function renderComponent() {
+    props.colors = colors;
+    tree = ReactDOM.render(<ColorSelector {...props} />, node);
+    component = TestUtils.findRenderedComponentWithType(tree, ColorSelector);
+    buttons = TestUtils.scryRenderedComponentsWithType(tree, AccessibleButton);
+  }
+
+  beforeEach(function() {
+    node = document.createElement('div');
+    tree = null;
+    component = null;
+    buttons = null;
+    colors = [ 'White', 'Blue', 'Magenta', 'Green', 'Yellow', 'Red' ];
+    props = {
+      enabled: true,
+      selectedColor: null,
+      colors: colors,
+      ariaLabel: 'ariaLabel',
+      onColorChange: function(color) {
+        props.selectedColor = color;
+      },
+      skinConfig: {
+        general: {
+          accentColor: 'blue'
+        }
+      }
+    };
+  });
+
+  it('should render a ColorSelector', function() {
+    renderComponent();
+    expect(component).toBeTruthy();
+    expect(buttons.length).toBe(6);
```

Comment:
Should we compare this against colors.length rather than 6?

Reply:
Updating